### PR TITLE
docs: Added some docs and githubpage generation

### PR DESCRIPTION
**Pull request by whynowy**
Signed-off-by: Derek Wang <whynowy@gmail.com>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ jobs:
         run: |
           IMAGE_NAMESPACE=${{ secrets.QUAYIO_ORG }} VERSION=${{ steps.version.outputs.VERSION }} DOCKER_PUSH=true make image-multi
 
-  bom:
+  Bom:
```
Comment (vigith) [inline .github/workflows/release.yml:90]: few places we use lower case, is this intended?

File: docs/QUICK_START.md [hunk 2]
```diff
@@ -1,29 +1,44 @@
...
+## A Simple Pipeline
+
+[Inter-Step Buffers](./INTER_STEP_BUFFER.md) are esstienal to run NumaFlow pipelines, to get Inter-Step Buffers in place, an [Inter-Step Buffer Service](./INTER_STEP_BUFFER_SERVICE.md) is required. We use [Nats JetStream](https://docs.nats.io/nats-concepts/jetstream) as a `Inter-Step Buffer Service` provider in this example.
+
 ```shell
 kubectl apply -f ./examples/0-isbsvc-jetstream.yaml
 ```
 
-After all the isbsvc pods are up, create a simple pipeline.
+After all the isbsvc pods are up, create a simple pipeline, and you can see the pipeline topoligy in the UI.
```
Comment (vigith) [inline docs/QUICK_START.md:40]:
```suggestion
After all the isbsvc pods are up, create a simple pipeline, and you can see the pipeline topology in the UI.
```

File: docs/QUICK_START.md [hunk 3]
```diff
@@ -1,29 +1,44 @@
...
-Create an `ISBSvc (Inter-Step Buffer Service)` object.
+Open a port-forward so you can access the UI on [https://localhost8443](https://localhost8443).
+
+```sh
+kubectl -n numaflow-system port-forward deployment/numaflow-server 8443:8443
+```
+
+## A Simple Pipeline
+
+[Inter-Step Buffers](./INTER_STEP_BUFFER.md) are esstienal to run NumaFlow pipelines, to get Inter-Step Buffers in place, an [Inter-Step Buffer Service](./INTER_STEP_BUFFER_SERVICE.md) is required. We use [Nats JetStream](https://docs.nats.io/nats-concepts/jetstream) as a `Inter-Step Buffer Service` provider in this example.
```
Comment (vigith) [inline docs/QUICK_START.md]:
```suggestion
[Inter-Step Buffers](./INTER_STEP_BUFFER.md) are essential to run NumaFlow pipelines, to get Inter-Step Buffers in place, an [Inter-Step Buffer Service](./INTER_STEP_BUFFER_SERVICE.md) is required. We use [Nats JetStream](https://docs.nats.io/nats-concepts/jetstream) as a `Inter-Step Buffer Service` provider in this example.
```

File: docs/QUICK_START.md [hunk 4]
```diff
@@ -42,20 +57,18 @@ Watch the `output` vertex pod log, you will see messages keep coming.
...
 
-To access the NumaFlow UI, use the following:
-
-```shell
-kubectl port-forward svc/numaflow-server-svc 8443
-```
-You should be able to access it at: https://localhost:8443
-
-You can delete the pipeline using
+You can delete the pipeline by using
```
Comment (vigith) [inline docs/QUICK_START.md:60]:
```suggestion
You can delete the pipeline by
```

for consistency with the next line

File: docs/QUICK_START.md [hunk 5]
```diff
@@ -42,20 +57,18 @@ Watch the `output` vertex pod log, you will see messages keep coming.
...
-The interstep buffer can be deleted by 
+The `Inter-Step Buffer Service` can be deleted by
+
 ```shell
 kubectl delete -f ./examples/0-isbsvc-jetstream.yaml
 ```
+
+## What's Next
+
+After exploring how a NumaFlow pipeline runs, you can check what data [Sources](./sources/GENERATOR.md) and [Sinks](./sinks/KAFKA.md) that NumaFlow supports, or learn how to write [User Defined Functions](./USER_DEFINED_FUNCTIONS.md).
```
Comment (vigith) [inline docs/QUICK_START.md:74]:
```suggestion
After exploring how a NumaFlow pipeline runs, you can check what data [Sources](./sources/GENERATOR.md) and [Sinks](./sinks/KAFKA.md) NumaFlow supports out of the box, or learn how to write [User Defined Functions](./USER_DEFINED_FUNCTIONS.md).
```

File: docs/RELEASES.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
-# Releasses
+# Releases
 
-You can find the most recent version under [Github release](https://github.com/numaproj/numaflow/releases).
+You can find the most recent version under [Github releases](https://github.com/numaproj/numaflow/releases).
```
Comment (vigith) [inline docs/RELEASES.md:3]:
```suggestion
You can find the most recent version under [Github Releases](https://github.com/numaproj/numaflow/releases).
```

for consistency

File: docs/USER_DEFINED_FUNCTIONS.md [hunk 7]
```diff
@@ -8,42 +8,37 @@ Data processing in the UDF is supposed to be idempotent.
 
 ## Builtin UDF
 
-There are some [builtin UDFs](./builtin-functions/) can be used directly.
+There are some `Builtin Functions` can be used directly.
```
Comment (vigith) [inline docs/USER_DEFINED_FUNCTIONS.md:11]:
```suggestion
There are some `Builtin Functions` that can be used directly.
```

File: docs/builtin-functions/README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+# Builtin Functions
+
+NumaFlow provides some builtin functions that can be used directly.
+
+**Cat**
+
+A `cat` builtin UDF does nothing but return the same messages it receives.
```
Comment (vigith) [inline docs/builtin-functions/README.md:7]:
```suggestion
A `cat` builtin UDF does nothing but return the same messages it receives. It is very useful for debugging and testing.
```

**Review by vigith — APPROVED** [review]
some NITs

File: .github/workflows/release.yml (hunk 1 above)
Comment (whynowy) [reply .github/workflows/release.yml:90]: To make all the first letter of github action job name capitalized :)

File: docs/QUICK_START.md (hunk 2 above)
Comment (whynowy) [reply docs/QUICK_START.md]: done.

File: docs/QUICK_START.md (hunk 3 above)
Comment (whynowy) [reply docs/QUICK_START.md]: done.

File: docs/QUICK_START.md (hunk 4 above)
Comment (whynowy) [reply docs/QUICK_START.md]: done.

File: docs/QUICK_START.md (hunk 5 above)
Comment (whynowy) [reply docs/QUICK_START.md]: done.

File: docs/RELEASES.md (hunk 6 above)
Comment (whynowy) [reply docs/RELEASES.md]: done

File: docs/USER_DEFINED_FUNCTIONS.md (hunk 7 above)
Comment (whynowy) [reply docs/USER_DEFINED_FUNCTIONS.md]: done.

File: docs/builtin-functions/README.md (hunk 8 above)
Comment (whynowy) [reply docs/builtin-functions/README.md]: done.

**Pull request merged**